documentation/docs/30-advanced/60-images.md

```diff
@@ -0,0 +1,149 @@
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+	plugins: [
++		enhancedImages(),
+		sveltekit()
+	]
+});
+```
+
+### Basic usage
+
+Use in your `.svelte` components by using `enhanced:img` rather than `img` and referencing the image file with a [Vite asset import](https://vitejs.dev/guide/assets.html#static-asset-handling) path:
+
+```svelte
+<enhanced:img src="./path/to/your/image.jpg" alt="An alt text" />
+```
+
+At build time, your `img` tag will be replaced with a `picture` providing multiple image types and sizes. It's only possible to downscale images without losing quality, which means that you should provide the highest resolution image that you need — smaller versions will be generated for the various device types that may request an image. If you're not using the [`sizes` attribute](#sveltejs-enhanced-img-srcset-and-sizes) you should provide your image at 2x resolution for HiDPI displays (a.k.a. retina displays).
+
+Since the `enhanced:img` tag will be replaced with a `picture` tag, you should refer to it as such within your `style` tag. Generally we recommend styling the `img` child of the `picture` tag rather than the `picture` tag itself to make it easier to switch between various solutions such as a dynamic CDN-based solution should you ever need to do so.
+
+### Dynamically choosing an image
+
+You can also manually import an image and then pass it to an `enhanced:img` tag. This is useful when you have a collection of static images and would like to dynamically choose one. You can create a collection of images manually [as we do on the homepage showcase](https://github.com/sveltejs/kit/blob/master/sites/kit.svelte.dev/src/routes/home/Showcase.svelte). In this case you will need to update both the `import` statement and `img` tag as shown below to indicate you'd like process them.
+
+```svelte
+<script>
+	import { MyImage } from './path/to/your/image.jpg?enhancedImg';
+</script>
+
+<enhanced:img src={MyImage} alt="An alt text" />
+```
+
+You can also use [Vite's `import.meta.glob`](https://vitejs.dev/guide/features.html#glob-import). Note that you will have to specify `enhancedImg` via a [custom query](https://vitejs.dev/guide/features.html#custom-queries):
+
+```js
+const pictures = import.meta.glob(
+	'/path/to/assets/*.{avif,gif,heif,jpeg,jpg,png,tiff,webp}',
+	{
+		query: {
+			enhancedImg: true
+		}
+	}
+);
+```
+
+### Intrinsic Dimensions
+
+`width` and `height` are optional as they can be inferred from the source image and will be automatically added when the `<enhanced:img>` tag is preprocessed. These attributes are added because the browser can reserve the correct amount of space when it knows image dimensions which prevents layout shift. If you'd like to use a different `width` and `height` you can style the image with CSS. Because the preprocessor adds a `width` and `height` for you, if you'd like one of the dimensions to be automatically calculated then you will need to specify that:
```

Comment:
```suggestion
`width` and `height` are optional as they can be inferred from the source image and will be automatically added when the `<enhanced:img>` tag is preprocessed. With these attributes, the browser can reserve the correct amount of space, preventing [layout shift](https://web.dev/articles/cls). If you'd like to use a different `width` and `height` you can style the image with CSS. Because the preprocessor adds a `width` and `height` for you, if you'd like one of the dimensions to be automatically calculated then you will need to specify that:
```